### PR TITLE
Delete item

### PR DESCRIPTION
**Pull request by rayhunglu**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -2,8 +2,11 @@ const router = require('express').Router()
 
 const home = require('./home')
 const { getUsers, postUser, loginUser } = require('./user')
-const { getItems, getItem, postItem, editItem } = require('./item')
+
+const { getItems, getItem, postItem, editItem, deleteItem } = require('./item')
 const { auth, authMid } = require('../middleware/auth')
+// const { checkToken } = require('../middleware/auth')
```
Comment (iglenwoo) [inline src/routes/index.js]: We can remove unnecessary lines

File: src/routes/index.js [hunk 2]
```diff
@@ -12,10 +15,18 @@ router.get('/users', getUsers)
 
 router.get('/items', getItems)
 router.get('/items/:id', getItem)
+// <<<<<<< delete-item
```
Comment (iglenwoo) [inline src/routes/index.js]: Same here

File: src/routes/index.js [hunk 3]
```diff
@@ -12,10 +15,18 @@ router.get('/users', getUsers)
...
 router.get('/chkToken', auth)
-router.get('/tokenTest', authMid, home)
+router.get('/checkTokenExp', checkToken, home)
+// =======
+// router.post('/items', authMid, postItem)
+// router.put('/items/:id', editItem)
+
+// router.get('/chkToken', auth)
+// router.get('/tokenTest', authMid, home)
+// >>>>>>> dev
```
Comment (iglenwoo) [inline src/routes/index.js]: Same here

File: src/routes/item/index.js [hunk 4]
```diff
@@ -57,9 +57,21 @@ const editItem = async function(req, res) {
   }
 }
 
+const deleteItem = async function(req, res) {
+  try {
+    const item = await Item.findById(req.params.id).exec()
+    item.delete()
```
Comment (iglenwoo) [inline src/routes/item/index.js]: I have multiple questions for this line.

1. I am wondering if `.delete()` works.
Mongoos support findOneAndDelete, deleteOne, and remove.
I referred to [this](https://stackoverflow.com/questions/42715591/mongodb-difference-remove-vs-findoneanddelete-vs-deleteone)

2. What does `delete()` return? If it returns a Promise, we want to use `await`.

3. Before this line, we would need to check if we found the item. If it does, we want to early return with an error.

File: src/server/index.js [hunk 5]
```diff
@@ -37,8 +37,16 @@ app.use(cookieParser())
 
 app.use(function(req, res, next) {
   res.header('Access-Control-Allow-Origin', 'http://localhost:3000')
+// <<<<<<< delete-item
+//   res.header("Access-Control-Allow-Methods", "GET,HEAD,OPTIONS,POST,PUT,DELETE");
+//   res.header(
+//     'Access-Control-Allow-Headers',
+//     'Origin, X-Requested-With, Content-Type, Accept'
+//   )
+// =======
```
Comment (iglenwoo) [inline src/server/index.js]: We want to remove unnecessary lines.

File: src/server/index.js [hunk 6]
```diff
@@ -37,8 +37,16 @@ app.use(cookieParser())
...
   res.header('Access-Control-Allow-Headers', '*')
   res.header('Access-Control-Allow-Methods', '*')
+// >>>>>>> dev
```
Comment (iglenwoo) [inline src/server/index.js]: Same here.

**Review by iglenwoo — CHANGES_REQUESTED** [review]
I left comments on where we want to change.
Also, it needs to fix conflicts again.

File: src/routes/item/index.js (hunk 4 above)
Comment (rayhunglu) [reply src/routes/item/index.js]: I changed the `delete` function to `deleteOne`.

File: src/routes/index.js [hunk 7]
```diff
@@ -14,7 +14,7 @@ router.get('/items', getItems)
 router.get('/items/:id', getItem)
 router.post('/items', authMid, postItem)
 router.put('/items/:id', editItem)
-
+router.delete('/items/:id', deleteItem)
```
Comment (iglenwoo) [inline src/routes/index.js]:
```suggestion
router.delete('/items/:id', authMid, deleteItem)
```
Oh! I forget to mention this. We should allow only the user who have valid token.
This change should work.
Just make sure this works to delete an item.

**Review by iglenwoo — CHANGES_REQUESTED** [review]
I added an additional change request which I forgot to mention.
After this change, this PR will be merged.

**Review by iglenwoo — APPROVED** [review]

**Pull request merged**